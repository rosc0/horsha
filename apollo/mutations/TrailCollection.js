import gql from 'graphql-tag'

// LIKE
export const TRAIL_LIKE = gql`
  mutation Like($trailId: TrailId!, $userId: ID) {
    trailMarkFavorite(id: $trailId, userId: $userId) {
      since
    }
  }
`

export const TRAIL_UNLIKE = gql`
  mutation unLike($trailId: TrailId!, $userId: ID) {
    trailUnmarkFavorite(id: $trailId, userId: $userId)
  }
`

// REVIEW
export const TRAIL_REVIEW_CREATE = gql`
  mutation trailReviewCreate(
    $trailId: TrailId!
    $body: BodyText
    $rating: Rating
    $trailPictureIds: [TrailPictureId!]
  ) {
    trailReviewCreate(
      input: {
        trailId: $trailId
        content: {
          body: $body
          rating: $rating
          trailPictureIds: $trailPictureIds
        }
      }
    ) {
      id
    }
  }
`

export const TRAIL_REVIEW_EDIT = gql`
  mutation trailReviewUpdate(
    $trailsReviewId: TrailReviewId!
    $body: BodyText
    $rating: Rating
    $trailPictureIds: [TrailPictureId!]
  ) {
    trailReviewUpdate(
      id: $trailsReviewId
      content: {
        body: $body
        rating: $rating
        trailPictureIds: $trailPictureIds
      }
    ) {
      id
    }
  }
`

export const TRAIL_REVIEW_DELETE = gql`
  mutation trailReviewDelete($trailsReviewId: TrailReviewId!) {
    trailReviewDelete(id: $trailsReviewId)
  }
`

export const TRAIL_PROFILE_EDIT = gql`
  mutation editTrail(
    $trailId: TrailId!
    $title: NameText!
    $description: BodyText
    $suitableForDriving: Boolean
    $surfaceTypes: [TrailSurfaceType!]
  ) {
    trailUpdateProfile(
      id: $trailId
      profile: {
        title: $title
        description: $description
        suitableForDriving: $suitableForDriving
        surfaceTypes: $surfaceTypes
      }
    ) {
      id
      title
    }
  }
`
