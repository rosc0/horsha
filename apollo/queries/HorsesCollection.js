import gql from 'graphql-tag'

export const GET_USER_HORSES = gql`
  query getUserHorses(
    $userId: UserId!
    $maxItems: BigInt = 10
    $cursor: String = null
  ) {
    horses(
      pagination: { maxItems: $maxItems, cursor: $cursor }
      filter: { memberUserId: $userId }
    ) {
      pageInfo {
        remaining
        previous
        next
      }
      collection {
        id
        name
        picture {
          url
        }
      }
    }
  }
`
