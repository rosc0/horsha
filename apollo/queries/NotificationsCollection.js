import gql from 'graphql-tag'

const NotificationCollectionFieldsFragment = gql`
  fragment NotificationCollectionFields on NotificationPage {
    pageInfo {
      remaining
      next
    }
    collection {
      type
      id
      dateUpdated
      dateRead
      subject {
        user {
          id
          name
          picture {
            url
          }
        }
        horse {
          id
          name
          picture {
            url
          }
        }
        post {
          text
          id
          author {
            id
            name
            picture {
              url
            }
          }
        }
        trail {
          id
          title
          creator {
            name
            picture {
              url
            }
          }
        }
      }
      target {
        name
        picture {
          url
        }
      }
      actors {
        collection {
          user {
            name
            picture {
              url
            }
          }
        }
      }
    }
  }
`

export const GET_NOTIFICATIONS_COLLECTION = gql`
  query notifications(
    $userId: UserId
    $maxItems: BigInt = 15
    $cursor: String = null
  ) {
    notifications(
      filter: { userId: $userId }
      pagination: { maxItems: $maxItems, cursor: $cursor }
    ) {
      ...NotificationCollectionFields
    }
  }
  ${NotificationCollectionFieldsFragment}
`
