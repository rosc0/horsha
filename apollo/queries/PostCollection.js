import gql from 'graphql-tag'

// export const PostCollectionFragment = gql`
//   fragment PostCollection on PostPage {
//     pageInfo {
//             remaining
//             previous
//      next
//     }
//     collection {
//       ...PostFragmentFields
//     }
//   }
//   ${PostCollectionFieldsFragment}
// `

// export const PostCollectionFieldsFragment = gql`
//   fragment PostFragmentFields on Post {
//     id
//     taggedHorse {
//       id
//     }
//     shareScope
//     text
//     media {
//       image {
//         url
//       }
//     }
//     author {
//       id
//       name
//       countryCode
//     }
//     dateCreated(format: RFC3339)
//     likes {
//       pageInfo {
//               remaining
// previous
// next
//       }
//       collection {
//         user {
//           name
//         }
//       }
//     }
//     likedSince
//     activity {
//       type
//       track {
//         distance
//       }
//     }
//     statusUpdate {
//       subject {
//         horse {
//           id
//         }
//         user {
//           id
//         }
//       }
//     }
//     publishState
//     datePublished
//     comments {
//       pageInfo {
// remaining
// previous
// next
//       }
//     }
//     canModify
//     canDelete
//     canUntagHorse
//     media {
//       image {
//         url
//       }
//     }
//   }
// `

// export const GET_POSTS_COLLECTION = gql`
//   query getPosts($maxItems: BigInt = 1) {
//     posts(
//       filter: { publishStates: PUBLISHED }
//       pagination: { maxItems: $maxItems }
//     ) {
//       ...PostCollectionFields
//     }
//   }
//   ${PostCollectionFragment}
// `

// export const GET_POSTS_FROM_USER_COLLECTION = gql`
//   query postsFromUser($maxItems: BigInt = 1, $authorUserId: String) {
//     posts(
//       filter: { authorUserId: $authorUserId }
//       pagination: { maxItems: $maxItems }
//     ) {
//       ...PostCollectionFields
//     }
//   }
//   ${PostCollectionFragment}
// `

// export const GET_POSTS_DRAFTS_COLLECTION = gql`
//   query getDraftPosts($maxItems: BigInt = 1) {
//     posts(
//       filter: { publishStates: UNPUBLISHED }
//       pagination: { maxItems: $maxItems }
//     ) {
//       ...PostCollectionFields
//     }
//   }
//   ${PostCollectionFragment}
// `

export const GET_POST_COMMENTS = gql`
  query getPostsComments(
    $postId: PostId!
    $maxItems: BigInt = 5
    $cursor: String = null
  ) {
    postComments(
      id: $postId
      pagination: { maxItems: $maxItems, cursor: $cursor }
    ) {
      pageInfo {
        remaining
        previous
        next
      }
      collection {
        text
        id
        canModify
        canRemove
        dateCreated
        author {
          id
          name
          picture {
            url
          }
        }
      }
    }
  }
`

export const PostCollectionFieldsNewsFeedFragment = gql`
  fragment PostFragmentFileds on Post {
    id
    taggedHorse {
      id
      name
      picture {
        url
      }
    }
    shareScope
    text
    media {
      id
      image {
        url
        width
        height
      }
      dateCreated
    }
    author {
      id
      name
      countryCode
      picture {
        url
      }
    }
    dateCreated
    likes {
      pageInfo {
        remaining
      }
      collection {
        user {
          name
        }
      }
    }
    likedSince
    activity {
      type
      trackAttachment {
        distance
      }
    }
    statusUpdate {
      subject {
        horse {
          id
          name
          picture {
            url
          }
        }
        user {
          id
          name
          picture {
            url
          }
        }
        trail {
          id
          title
          fullWaypointsUrl
          description
          dateCreated(format: UNIX)
          reviews {
            pageInfo {
              remaining
            }
            collection {
              id
              author {
                id
                name
              }
              rating
              body
              dateCreated(format: UNIX)
              dateLastUpdated(format: UNIX)
            }
          }
          creator {
            id
            name
          }
          review {
            author {
              id
              name
            }
          }
          previewWaypointsUrl
          itinerary
          suitableForDriving
          distance
          favoriteSince
          ratingAverage
          canModify
          canDelete
          pictures(maxItems: 5) {
            pageInfo {
              remaining
            }
            collection {
              id
              image {
                url
              }
              author {
                name
              }
            }
          }
          elevationSummary {
            totalAscent
            totalDescent
            minAltitude
            maxAltitude
          }
          boundingBox {
            topRight {
              longitude
              latitude
            }
            bottomLeft {
              longitude
              latitude
            }
          }
          surfaceTypes
          poiKinds
          poiOccurrences {
            kind
            count
          }
          pois {
            collection {
              distanceFromStart
              poi {
                kind
                creator {
                  name
                  id
                }
                location {
                  latitude
                  longitude
                }
                description
              }
            }
          }
        }
      }
      type
    }
    publishState
    datePublished
    comments {
      pageInfo {
        remaining
      }
      collection {
        id
        text
      }
    }
    canModify
    canUntagHorse
  }
`
