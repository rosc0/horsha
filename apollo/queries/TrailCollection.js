import gql from 'graphql-tag'

export const TrailCollectionFieldsFragment = gql`
  fragment TrailCollectionFields on TrailPage {
    collection {
      id
      title
      description
      creator {
        id
        name
      }
      previewWaypointsUrl
      fullWaypointsUrl
      itinerary
      suitableForDriving
      distance
      favoriteSince
      ratingAverage
      canModify
      canDelete
      reviews {
        pageInfo {
          remaining
        }
        collection {
          author {
            id
            name
          }
          rating
        }
      }
      pictures {
        collection {
          image {
            url
          }
        }
      }
      elevationSummary {
        totalAscent
        totalDescent
      }
      boundingBox {
        topRight {
          longitude
          latitude
        }
        bottomLeft {
          longitude
          latitude
        }
      }
      poiKinds
    }
    pageInfo {
      remaining
      next
      previous
    }
  }
`

export const GET_TRAILS_COLLECTION = gql`
  query getTrails(
    $filter: TrailFilterInput
    $maxItems: BigInt = 10
    $cursor: String = null
  ) {
    trails(
      filter: $filter
      pagination: { maxItems: $maxItems, cursor: $cursor }
    ) {
      ...TrailCollectionFields
    }
  }
  ${TrailCollectionFieldsFragment}
`

export const GET_TRAIL_BY_ID = gql`
  query getTrail($trailId: TrailId!) {
    trail(id: $trailId) {
      id
      title
      fullWaypointsUrl
      description
      dateCreated(format: UNIX)
      reviews {
        pageInfo {
          remaining
        }
        collection {
          id
          author {
            id
            name
          }
          rating
          body
          dateCreated(format: UNIX)
          dateLastUpdated(format: UNIX)
        }
      }
      creator {
        id
        name
      }
      review {
        author {
          id
          name
        }
      }
      previewWaypointsUrl
      itinerary
      suitableForDriving
      distance
      favoriteSince
      ratingAverage
      canModify
      canDelete
      pictures(maxItems: 5) {
        pageInfo {
          remaining
        }
        collection {
          id
          image {
            url
          }
          author {
            name
          }
        }
      }
      elevationSummary {
        totalAscent
        totalDescent
        minAltitude
        maxAltitude
      }
      boundingBox {
        topRight {
          longitude
          latitude
        }
        bottomLeft {
          longitude
          latitude
        }
      }
      surfaceTypes
      poiKinds
      poiOccurrences {
        kind
        count
      }
      pois {
        collection {
          distanceFromStart
          poi {
            kind
            creator {
              name
              id
            }
            location {
              latitude
              longitude
            }
            description
          }
        }
      }
    }
  }
`

export const GET_TRAIL_PICTURES = gql`
  query trailPictures($trailId: TrailId!) {
    trailPictures(filter: { trailId: $trailId }) {
      pageInfo {
        remaining
        next
        previous
      }
      collection {
        id
        image {
          url
          width
          height
        }
        author {
          id
        }
      }
    }
  }
`

export const GET_PREVIEW_TRAIL_PICTURES = gql`
  query trailPictures($trailId: TrailId!) {
    trailPictures(filter: { trailId: $trailId }) {
      pageInfo {
        remaining
        next
        previous
      }
      collection {
        id
        image {
          url
        }
      }
    }
  }
`

export const GET_TRAIL_REVIEWS = gql`
  query getTrailReviews(
    $trailId: TrailId!
    $maxItems: BigInt = 10
    $cursor: String = null
  ) {
    trailReviews(
      pagination: { maxItems: $maxItems, cursor: $cursor }
      filter: { trailId: $trailId }
    ) {
      pageInfo {
        remaining
        next
        previous
      }
      collection {
        id
        author {
          id
          name
        }
        canDelete
        rating
        canModify
        dateCreated(format: UNIX)
        dateLastUpdated(format: UNIX)
        body
      }
    }
  }
`
