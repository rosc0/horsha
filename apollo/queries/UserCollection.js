import gql from 'graphql-tag'

export const UserFieldsFragment = gql`
  fragment UserFields on User {
    id
    name
    picture {
      url
    }
    countryCode
  }
`

// export const CurrentUserFieldsFragment = gql`
//   fragment CurrentUserFields on User {
//     id
//     name
//     picture {
//       url
//     }
//     countryCode
//     dateCreated
//     about
//     birthday
//     ridingTotals {
//       ascent
//       descent
//       distance
//       duration
//       count
//     }
//     account {
//       preferences {
//         unitSystem
//         heightUnit
//         pushEnabled
//         unitSystem
//         pushTopics {
//           horse
//           post
//         }
//       }
//     }
//   }
// `

export const UserFollowCollectionFieldsFragment = gql`
  fragment UserFollowCollectionFields on UserPage {
    pageInfo {
      next
      previous
      remaining
    }
    collection {
      id
      name
      picture {
        url
      }
      countryCode
      inboundFriendRequestState
    }
  }
`

// # query me {
//   #   userMe {
//   #     ...CurrentUserFields
//   #   }
//   # }
//   # ${CurrentUserFieldsFragment}

export const CURRRENT_USER = gql`
  query userMe {
    userMe {
      id
      name
      picture {
        url
      }
      countryCode
      dateCreated
      about
      birthday
      ridingTotals {
        ascent
        descent
        distance
        duration
        count
      }
      account {
        primaryEmail
        registeredEmails {
          email
          confirmed
        }
        preferences {
          unitSystem
          heightUnit
          pushEnabled
          unitSystem
          pushTopics {
            horse
            post
          }
        }
      }
    }
  }
`

export const GET_FOLLOWERS = gql`
  query getFollowerFromUserId(
    $userId: UserId!
    $maxItems: BigInt = 15
    $cursor: String = null
  ) {
    userList(
      filter: { followingUserId: $userId }
      pagination: { maxItems: $maxItems, cursor: $cursor }
    ) {
      ...UserFollowCollectionFields
    }
  }
  ${UserFollowCollectionFieldsFragment}
`

export const GET_FOLLOWING = gql`
  query getMyFollowingList(
    $userId: UserId!
    $maxItems: BigInt = 15
    $cursor: String = null
  ) {
    userList(
      filter: { followeeUserId: $userId }
      pagination: { maxItems: $maxItems, cursor: $cursor }
    ) {
      ...UserFollowCollectionFields
    }
  }
  ${UserFollowCollectionFieldsFragment}
`

export const GET_FRIENDS = gql`
  query getMyFriendsList(
    $userId: UserId!
    $maxItems: BigInt = 15
    $cursor: String = null
  ) {
    userList(
      filter: { friendsWithUser: $userId }
      pagination: { maxItems: $maxItems, cursor: $cursor }
    ) {
      ...UserFollowCollectionFields
    }
  }
  ${UserFollowCollectionFieldsFragment}
`

export const GET_PENDING_INVITATIONS = gql`
  query getPendingInvitations(
    # $userId: UserId!
    $maxItems: BigInt = 15
    $cursor: String = null
  ) {
    userList(
      filter: { inboundFriendRequestStates: [PENDING] }
      pagination: { maxItems: $maxItems, cursor: $cursor }
    ) {
      ...UserFollowCollectionFields
    }
  }
  ${UserFollowCollectionFieldsFragment}
`

export const GET_IGNORED_INVITATIONS = gql`
  query getIgnoredInvitations(
    # $userId: UserId!
    $maxItems: BigInt = 15
    $cursor: String = null
  ) {
    userList(
      filter: { inboundFriendRequestStates: [IGNORED] }
      pagination: { maxItems: $maxItems, cursor: $cursor }
    ) {
      ...UserFollowCollectionFields
    }
  }
  ${UserFollowCollectionFieldsFragment}
`

export const GET_PEOPLE_YOU_MAY_KNOW = gql`
  query getPeopleYouMayKnow(
    $userId: UserId!
    $maxItems: BigInt = 15
    $cursor: String = null
  ) {
    userFriendSuggestions(
      id: $userId
      pagination: { maxItems: $maxItems, cursor: $cursor }
    ) {
      pageInfo {
        next
        previous
        remaining
      }
      collection {
        user {
          id
          name
          picture {
            url
          }
          countryCode
        }
      }
    }
  }
`
