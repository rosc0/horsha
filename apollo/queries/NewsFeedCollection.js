import gql from 'graphql-tag'
import { PostCollectionFieldsNewsFeedFragment } from './PostCollection'
// import { PostCollectionFieldsFragment } from './PostCollection'

// // export const GET_NEWSFEED_COLLECTION = gql`
// //   query newsFeed($maxItems: BigInt = 1, $userId: UserId!) {
// //     newsFeed(userId: $userId, pagination: { maxItems: $maxItems }) {
// //       ...NewsFeedItemCollectionFields
// //     }
// //   }
// //   ${NewsFeedItemCollectionFieldsFragment}
// // `

// const NewsFeedItemCollectionFieldsFragment = gql`
//   fragment NewsFeedItemCollectionFields on NewsFeedItemPage {
//     pageInfo {
//       remaining
//     }
//     collection {
//       id
//       targetUser {
//         name
//       }
//       visible
//       type
//       dateCreated
//       subject {
//         ...PostFragment
//       }
//     }
//   }
//   ${PostFragment}
// `

// const PostFragment = gql`
//   fragment PostFragment on NewsSubject {
//     post {
//       id
//     }
//   }
//   # ${PostCollectionFieldsFragment}
// `

// export const GET_NEWSFEED_COLLECTION = gql`
//   query newsFeed($maxItems: BigInt = 1, $userId: UserId!) {
//     newsFeed(
//       userId: $userId,
//       pagination: { maxItems: $maxItems }
//     ) {
//       pageInfo {
//         remaining
//       }
//       collection {
//         id
//         targetUser {
//           name
//         }
//         visible
//         type
//         dateCreated
//         subject {
//           post {
//             id
//           }
//         }
//       }
//     }
//   }
// `

// fragment PostFragment on NewsSubject {
//   post {
//     ...PostFragmentFileds
//   }
// }

const PostFragmentFragment = gql`
  fragment PostFragment on NewsSubject {
    post {
      ...PostFragmentFileds
    }
  }
  ${PostCollectionFieldsNewsFeedFragment}
`

const NewsFeedItemCollectionFieldsFragment = gql`
  fragment NewsFeedItemCollectionFields on NewsFeedItemPage {
    pageInfo {
      remaining
      previous
      next
    }
    collection {
      id
      targetUser {
        name
      }
      visible
      type
      dateCreated
      subject {
        ...PostFragment
      }
    }
  }
  ${PostFragmentFragment}
`

export const GET_NEWSFEED_COLLECTION = gql`
  query newsFeed(
    $maxItems: BigInt = 15
    $userId: UserId!
    $cursor: String = null
  ) {
    newsFeed(
      userId: $userId
      pagination: { maxItems: $maxItems, cursor: $cursor }
    ) {
      ...NewsFeedItemCollectionFields
    }
  }
  ${NewsFeedItemCollectionFieldsFragment}
`

export const GET_NEWSFEED_ITEM = gql`
  query newsFeedItem($postId: PostId!) {
    post(id: $postId) {
      ...PostFragmentFileds
    }
  }
  ${PostCollectionFieldsNewsFeedFragment}
`
