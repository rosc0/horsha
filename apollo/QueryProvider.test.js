import 'react-native'
import React from 'react'
import { MockedProvider } from 'react-apollo/test-utils'
import renderer from 'react-test-renderer'

import Trails from './QueryProvider'
import { GET_TRAILS_COLLECTION } from './queries/TrailCollection'

const wait = (amount = 0) => new Promise(resolve => setTimeout(resolve, amount))

const variables = {
  maxItems: 5,
  filter: {
    boundingBox: {
      topRight: {
        latitude: 50.233002419847715,
        longitude: 6.50906324568013,
      },
      bottomLeft: {
        latitude: 50.21685451906035,
        longitude: 6.098913991729163,
      },
    },
  },
}

const responseExpec = {
  data: {
    trails: {
      collection: [
        {
          id: 'ed3d54ed-f0f0-433a-9ed2-b1d60f9a2782',
          title:
            'Naar rolfilm GHB. Chhb cfh ghj ghjj ghhhh hhhhhgg vgghh ghhhhj ghhhhh hhghgg hhghgg',
          description: 'New description another bcjcj',
          __typename: 'Trail',
        },
        {
          id: 'd177923d-378b-471e-8f51-c6704e948f66',
          title: 'eifel hele trektocht',
          description: null,
          __typename: 'Trail',
        },
        {
          id: 'c519c9d1-2604-4e72-954b-968378b1538e',
          title: 'Prum to Schonfeld',
          description: null,
          __typename: 'Trail',
        },
        {
          id: 'ac677341-1a01-4126-8f36-21e0ac82d364',
          title: 'Naar rolf ',
          description: null,
          __typename: 'Trail',
        },
      ],
      __typename: 'TrailCollection',
    },
  },
}

it('should render without error', () => {
  renderer.create(
    <MockedProvider mocks={[]}>
      <Trails />
    </MockedProvider>
  )
})

it('should render loading state initially', () => {
  const component = renderer.create(
    <MockedProvider mocks={[]}>
      <Trails />
    </MockedProvider>
  )
  expect(component.toJSON()).toMatchSnapshot()
  expect(component.toJSON().children[0].children[0]).toContain('Loading...')
})

it('should show data', async () => {
  const mock = {
    request: {
      query: GET_TRAILS_COLLECTION,
      variables: variables,
    },
    result: responseExpec,
  }

  const component = renderer.create(
    <MockedProvider mocks={[mock]} addTypename={false}>
      <Trails />
    </MockedProvider>
  )

  await wait(0) // wait for response

  // const tree = component.toJSON()
  // expect(tree).toMatchSnapshot()
  // expect(tree.children[0].children[0]).toContain('Error :(')
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
  // const p = component.root.findByType('View');
  // expect(p.children).toContain('Buck is a poodle');
})

it('should show error UI', async () => {
  const mock = {
    request: {
      query: GET_TRAILS_COLLECTION,
      variables: variables,
    },
    error: new Error('aw shucks'),
  }

  const component = renderer.create(
    <MockedProvider mocks={[mock]} addTypename={false}>
      <Trails />
    </MockedProvider>
  )

  await wait(0) // wait for response

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
  expect(tree.children[0].children[0]).toContain('Error :(')
})
