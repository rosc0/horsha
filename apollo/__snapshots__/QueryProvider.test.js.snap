// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render loading state initially 1`] = `
<View>
  <Text>
    Loading...
  </Text>
</View>
`;

exports[`should show data 1`] = `
<View>
  <Text>
    Error :(
  </Text>
</View>
`;

exports[`should show error UI 1`] = `
<View>
  <Text>
    Error :(
  </Text>
</View>
`;
